### ISSUE TITLE
ForbiddenMethodCall does not trigger on default interface implementations

### ISSUE THREAD

**Issue by vatbub** [detekt/detekt#6474]
This bug is related to #4204

## Expected Behavior
Consider the following code:

```kotlin
import javafx.beans.property.SimpleObjectProperty

class DetektForbiddenMethodCallDefaultInterfaceDemo {
    val objectProperty = SimpleObjectProperty("Hello there")
    val mappedProperty = objectProperty.map { it.uppercase() }
}
```

With the following configuration, ForbiddenMethodCall should flag the call to `.map()`.

```yml
  ForbiddenMethodCall:
    active: true
    methods:
      - reason: 'The javafx-built in ObservableValue.map function handles null values in an unusual and in Kotlin undesired way. Use map2() instead.'
        value: 'javafx.beans.value.ObservableValue.map'
```

## Observed Behavior
The call is not flagged. However, the following configuration will flag the call:

```yml
  ForbiddenMethodCall:
    active: true
    methods:
      - reason: 'The javafx-built in ObservableValue.map function handles null values in an unusual and in Kotlin undesired way. Use map2() instead.'
        value: 'javafx.beans.value.ObservableValue.map'
      - reason: 'The javafx-built in ObservableValue.map function handles null values in an unusual and in Kotlin undesired way. Use map2() instead.'
        value: 'javafx.beans.property.SimpleObjectProperty.map'
```

It seems like this is the same bug as #4204 except with one difference: `SimpleObjectProperty` does not override `map`. Instead, the `ObservableValue` interface contains a default implementation, which apparently is not recognized by the patch that fixed #4204 .

Here are the relevant portions of source code for reference: [ObservableValue](https://github.com/openjdk/jfx/blob/eb7de72dafecbedc83c2215b6aed7432d4ec80f9/modules/javafx.base/src/main/java/javafx/beans/value/ObservableValue.java#L178) and [SimpleObjectProperty](https://github.com/openjdk/jfx/blob/eb7de72dafecbedc83c2215b6aed7432d4ec80f9/modules/javafx.base/src/main/java/javafx/beans/property/SimpleObjectProperty.java#L39)

## Steps to Reproduce
Create a project with the following dependency:

```xml
<dependency>
    <groupId>org.openjfx</groupId>
    <artifactId>javafx-base</artifactId>
    <version>19.0.2.1</version>
</dependency>
```

Then paste the above code in the project and run detekt with type resolution enabled on that project.

## Your Environment
* Version of detekt used: 1.23.1
* Version of Gradle used (if applicable): N/A
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: Windows 10
* Link to your project (if it's a public repository): private

**Comment by atulgpt** [comment]
Hi, @vatbub I have added TC for https://github.com/detekt/detekt/pull/6475 but in those TCs detekt is working as expected. Can you check if you have used the right `detekt` config?

**Comment by vatbub** [comment]
As I said, I have type resolution enabled and `ForbiddenMethodCall` does flag the method call if I configure the explicit implementation. Can you point me to what could be wrong with my config?

**Comment by atulgpt** [comment]
Hi, @vatbub can you check if my PR at https://github.com/detekt/detekt/pull/6475/files correctly tests your scenarios or not Also can you cross-check if the config has the correct package name or not for `ObservableValue.map` method?

**Comment by vatbub** [comment]
Hey, yes, I double-checked the package names. Could the reason maybe also be that the relevant javafx-classes are written in Java and your unit tests are written in Kotlin?

**Comment by vatbub** [comment]
Hmm, I just checked the standard library myself and your test case should actually cover it. I don't know what else could be different between your test case and my use case. Have you tried recreating my use case with a sample project?

**Comment by vatbub** [comment]
To make sure that this was not a configuration issue in my project, I created a minimal sample project to reproduce the issue and shared it on GitHub: 

https://github.com/vatbub/DetektForbiddenMethodCallDefaultInterfaceDemo

The issue still seems to persist there, so I don't think that it's a configuration issue. Let me know if you need more help .
Cheers :)

**Comment by atulgpt** [comment]
Hi @vatbub I have raised one PR which should fix the issue. @BraisGabin is there as way that PR can be tried earlier as well?

**Comment by BraisGabin** [comment]
> @BraisGabin is there a way that PR artifacts can be tried earlier as well?

No. An alternative is to checkout detekt and run it as a include build in the target project. But with main pointing to 2.0.0 that would be difficult. Anyway I think that your fix has a lot of sense.

**Comment by vatbub** [comment]
I cherry-picked both of your PRs on top of the v1.23.1 tag, compiled it locally and then used that jar in my sample project. Your last PR seems to fix the issue successfully. Thanks :)

**Comment by vatbub** [comment]
FYI I just uploaded the patch to my sample project if you want so check it out as well.
